socorro/processor/processor_2015.py

```diff
@@ -279,11 +276,72 @@ def _log_job_end(self, success, crash_id):
         )
 
     def close(self):
-        for a_rule_set_name, a_rule_set in self.rule_system.iteritems():
-            self.config.logger.debug('closing %s', a_rule_set_name)
+        self.config.logger.debug('done closing rules')
+        for rule in self.rules:
             try:
-                a_rule_set.close()
+                self.config.logger.debug('trying to close %s',
+                                         rule.__class__)
+                close_method = rule.close
             except AttributeError:
-                # guess we don't need to close that rule
-                pass
-        self.config.logger.debug('done closing rules')
+                self.config.logger.debug('%s has no close',
+                                         rule.__class__)
+                # no close method mean no need to close
+                continue
+            close_method()
```

Comment:
I think instead of this, I'd add a vestigal `.close()` to `Rule` that's a no-op.